lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py

```diff
@@ -1105,21 +1105,22 @@ def wait_for_restart_to_finish(self, wait_time):
         Args:
             wait_time: (int) max time to wait for services to become active
         """
-        print(
-            f"Waiting for a maximum of {wait_time} "
-            f"seconds for restart to finish",
-        )
-        if self._init_system == InitMode.DOCKER:
+        if self._init_system in (InitMode.DOCKER, InitMode.SYSTEMD):
+            print(
+                f"Waiting for a maximum of {wait_time} "
+                f"seconds for restart to finish",
+            )
             start_time = time.time()
             all_services_active = False
             while (
                 not all_services_active
-                and time.time() - start_time < wait_time
             ):
                 all_services_active = self.check_if_magma_services_are_active()
                 time.sleep(5)
-        elif self._init_system == InitMode.SYSTEMD:
-            time.sleep(wait_time)
+
+                if time.time() - start_time > wait_time:
+                    print("Timeout reached while waiting for docker service to restart")
```

Comment:
```suggestion
                    print("Timeout reached while waiting for services to restart")
```
I think this gets executed in both cases?

Reply:
The message send in this line has to be received by the mme within 32s after the `self._s1ap_wrapper.sendPdnConnectivityReq(ue_id, apn)` has been send:
[retransmitted_response = self._s1ap_wrapper.s1_util.get_response()](https://github.com/magma/magma/blob/7f4c4fed5286122113b85f287f7c887540c6b611/lte/gateway/python/integ_tests/s1aptests/test_3485_timer_for_default_bearer_with_mme_restart.py#L124)

The old code was always waiting at least 30s for the mme to restart so in cases where the mme restarted quickly and the test was running slowly the bearer would be terminated due to timeout.

Reply:
I guess this is the answer for my other question. :)